engineering_tools/rosbag_matching/rosbag_match

```diff
@@ -0,0 +1,66 @@
+
+#!/bin/bash 
+
+rosbag=$1
+logfile=$2
+
+# get the bag time from rosbag info
+while read -r line ; do
+
+	if [[ $line =~ \([0-9]+\.[0-9]+\) ]]; then
+		IFS=" "
+		read -ra bagtime <<< "$line"
+		ros_time=$(echo "${bagtime[5]:1:13}" | bc -l)
+		break
+	fi
+
+done < <(rosbag info $rosbag) 
+
+
+# go inside logfiles
+cd $logfile
+
+
+match="0"
+# go through every log directory in logfiles
+for dir in */; do
+	cd $dir
+	echo Log Directory:
+	echo $dir
+	while read -r line ; do
+		if [[ $line =~ [0-9]+\.[0-9]+ ]]; then
+			# get the log time from rosout.log
+			IFS=" "
+			read -ra time <<< "$line"
+			
+			# determine match
+			difference=$(echo "${time[0]:0:13}-$ros_time" | bc -l)
+#			echo $difference
+			if [[ "$(echo "$difference<0 " | bc -l)" -eq "1"  ]]; then
+				difference=${difference:1}					
+			fi
+			echo Log Time: ${time:0:13} ; echo Rosbag Time: $ros_time ; echo
+			if [[ "$(echo "$difference<60" | bc -l)" -eq "1"  ]]; then
+				echo "Match!" ; echo
+				match="1"
+				match_dir=$dir
+				match_time=${time:0:13}
+				match_diff=$difference
+								
+				# rename the directory
+				cd ..
+				match_log=${rosbag%.bag}.log
```

Comment:
I don't like the ".log" extension for the name of a directory.  Text log files inside this dir are already named *.log.  Also, directory names typically do not have a . extension, so I think it would be very confusing.  I suggest a suffix of "_logs" instead.